packages/docs/getting-started/running-a-validator.md

```diff
@@ -46,33 +66,227 @@ The code you'll see on this page is bash commands and their output.
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account new"
+docker run -v $PWD/proxy:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account new"
+```
+
+Those commands will prompt you for a passphrase, ask you to confirm it, and then will output your account address: `Address: {<YOUR-ACCOUNT-ADDRESS>}`
+
+Let's save these addresses to environment variables, so that you can reference it later (don't include the braces):
+
+```bash
+export CELO_VALIDATOR_GROUP_ADDRESS=<YOUR-VALIDATOR-GROUP-ADDRESS>
+export CELO_VALIDATOR_ADDRESS=<YOUR-VALIDATOR-ADDRESS>
+export CELO_PROXY_ADDRESS=<YOUR-PROXY-ADDRESS>
+```
+
+In order to register the validator later on, generate a "proof of possession" - a signature proving you know your validator's BLS private key. Run this command:
+
+```bash
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account proof-of-possession $CELO_VALIDATOR_ADDRESS"
+```
+
+It will prompt you for the passphrase you've chosen for the validator account. Let's save the resulting proof-of-possession to an environment variable:
+
+```bash
+export CELO_VALIDATOR_POP=<YOUR-VALIDATOR-PROOF-OF-POSSESSION>
+```
+
+#### Deploy the validator and proxy nodes
+
+We initialize the docker containers for the validator and the proxy, building from an image for the network and initializing Celo with the genesis block:
+
+```bash
+docker run -v $PWD/proxy:/root/.celo $CELO_IMAGE:$CELO_NETWORK init /celo/genesis.json
+docker run -v $PWD/validator:/root/.celo $CELO_IMAGE:$CELO_NETWORK init /celo/genesis.json
+```
+
+To participate in consensus, we need to set up our nodekey for our accounts. We can do so via the following commands \(it will prompt you for your passphrase\):
+
+```bash
+docker run -v $PWD/proxy:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account set-node-key $CELO_PROXY_ADDRESS"
+docker run -v $PWD/validator:/root/.celo --entrypoint /bin/sh -it $CELO_IMAGE:$CELO_NETWORK -c "geth account set-node-key $CELO_VALIDATOR_ADDRESS"
+```
+
+In order to allow the node to sync with the network, give it the address of existing nodes in the network:
+
+```bash
+docker run -v $PWD/proxy:/root/.celo --entrypoint cp $CELO_IMAGE:$CELO_NETWORK /celo/static-nodes.json /root/.celo/
+docker run -v $PWD/validator:/root/.celo --entrypoint cp $CELO_IMAGE:$CELO_NETWORK /celo/static-nodes.json /root/.celo/
+```
+
+At this point we are ready to start up the proxy:
+
+```bash
+docker run --name celo-proxy -p 8545:8545 -p 8546:8546 -p 30303:30303 -p 30303:30303/udp -p 30503:30503 -p 30503:30503/udp -v $PWD/proxy:/root/.celo $CELO_IMAGE:$CELO_NETWORK --verbosity 3 --networkid $NETWORK_ID --syncmode full --rpc --rpcaddr 0.0.0.0 --rpcapi eth,net,web3,debug --maxpeers 1100 --etherbase=$CELO_PROXY_ADDRESS --proxy.proxy --proxy.proxiedvalidatoraddress $CELO_VALIDATOR_ADDRESS --proxy.internalendpoint :30503
```

Comment:
We should add `--restart=Always` so that if the docker container crashes (a real possibility due to the memleak in geth) it'll be automatically restarted

Also, this will run it in the foreground, a note about using `-d` to run in background could be useful to those unfamiliar with Docker. But then they won't be able to see logs as easily, so I'm not super passionate

Reply:
:+1: